### ISSUE TITLE
Improve Player UI

### ISSUE THREAD

**Issue by TobiGr** [TeamNewPipe/NewPipe#1317]
At the moment our player UI is quite untidy and not easy to use on small devices. In addition, the symbol for additional actions is very small and therefore difficult to reach.
My suggestion to make more options accessible is a second overlay which can be opened via the settings icon in the right corner. Tapping on an icon on that screen should open another instance with more options if necessary. Tapping on free space should close the menu.
I added a placeholder for Pipecast or any other feature (equalizer, ...) so we won't have to spend time on restructuring it again in the near future.
I thought about the problem with the tiny popup which has even less space to work with. Maybe we should just open NewPipe to adjust the popup settings and close it afterwards or auto-resize it until the menu gets closed.
@theScrabi @karyogamy @mauriciocolli @wb9688 Is there anything else we need to add to the player?

![player_ui_new](https://user-images.githubusercontent.com/17365767/38772923-161b5e18-4042-11e8-9c4e-da01fb7f24d7.png)
![player_ui_options](https://user-images.githubusercontent.com/17365767/38772924-1637510e-4042-11e8-9874-c020048803d5.png)
![player_ui_subtitles](https://user-images.githubusercontent.com/17365767/38773305-3bd3de76-4049-11e8-9b61-8eba07b05b89.png)

**Comment by karyogamy** [comment]
I'd prefer something similar to VLC's layout, where the play/pause, previous, next selections are dropped to the bottom, creating something similar to our background player queue. Where these three buttons are surrounded by 3 other most used functionalities and an overflow button, which less used buttons are shown on a translucent dialog.

This way, we drop all button access to the bottom of the layout, making control easier on the hand and fingers. 

Also, we should merge all track (audio/video/subtitle) track selection to a single dialog, similar to the ExoPlayer demo app. After the player is refactored, swapping between the popup and main players should be fast enough that we don't really need to have many UI elements on the popup and can just change/inherit the settings from the main player.

**Comment by justanidea** [comment]
With so many buttons isnt it easier for the user to swipe right to left and see a pannel with all of it?

**Comment by justanidea** [comment]
Because we dont have so many place on top & bottom, (bottom is already crowded by subtitles is used),  but we have much more space on the sides 

Ps: swaping is more handy for people with little phone, than clicking on a tiny button  which gives tiny others... And it makes less steps to go to it

**Comment by FlorianSteenbuck** [comment]
Implemented as own player with lower display resolution. Would be the solution to this. And through the swiping for this player away.

**Comment by phnikola** [comment]
2 more suggestions on the UI front:
- video lists look cluttered on larger device (Galaxy Note 5, default DPI), would welcome more cozy spread between lines related to one video
- notification controls - look outdated. Check Simple Mobile Tools - Music Player. https://play.google.com/store/apps/details?id=com.simplemobiletools.musicplayer ... Unsure which library they used for notifications - but I guess its free, as this provider makes free apps only.

**Comment by Stypox** [comment]
@avently said in #4456:

#### Describe the feature you want
Right now we have to select arrow down button in order to see second line of buttons like subtitles, aspect ratio, open in browser, etc. This is fine until someone will want to show more buttons inside this line. All the buttons from second line should be replaced with AlertDialog or BottomSheet or Popup menu with enabled icons. I'm not sure which one is the best, needs testing to be sure. Popup menu allows to show submenu for subtitles and aspect ratio which is useful.

By making this change we can leave arrow down button but make it working like a scroller to the description. This will allow to tap on it and to see a description even in fullscreen. Once someone make a swipe-to-go-to-description even this button can be removed.

So the proposed look of the buttons in non-fullscreen:
Close video ------------ quality | speed | setup

In fullscreen:
 ------------ quality | speed | setup | arrow down

If the `setup` button will be a three dots icon it should be the last in a row in landscape too.

#### Is your feature request related to a problem? Please describe it


The problem is there are to many buttons. Some users want to add even more buttons. I don't think this UI element like second line with buttons is a great element in the case of many buttons.

#### How will you/everyone benefit from this feature?


We can close many issues after this idea is implemented. Also the player will be much faster in case of view relayout because it will lose some buttons.

**Comment by avently** [comment]
Not sure how other devs will find this old issue, but anyway. What you guys think about the idea to replace second line of buttons with dialog or something similar?

**Comment by opusforlife2** [comment]
A dark transparent version of what is in the OP would be my preference.

**Comment by Stypox** [comment]
@avently I agree with you and @opusforlife2, the current layout looks good with a small number of buttons, but already does not feel completely right. I might work on this in the future, though not for 0.20.1, as it would take some time and thoughts

**Comment by Angelk90** [comment]
@TobiGr, @karyogamy, @FlorianSteenbuck, @CharlieBrown12, @Stypox, @avently, @opusforlife2: 

I wanted to open a discussion, but it has not yet been decided whether to enable them or not.
The images are based on version 0.21.9

I would recommend a ui similar to that of the youtube player.

1) The current timer of the playing video and the total time of the video, I would put them close together.
As seen in the youtube image.

2) For changing video quality, playback speed it would be better to put them in a bottomsheet menu.

3) I would put a button on the right to enable subtitles.

4) A button to open the bottomsheet menu, where the following characteristics should be entered:
- Video quality
- Loop the video
- Choice of subtitle to use
- Playback speed

5) I would remove the following features:
- The "x" on the left to exit the video player
- World icon that allows you to open the video in youtube app
- Share it
- The button to mute audio

Youtube:
![image](https://user-images.githubusercontent.com/20476002/129247020-8d7663f3-95a5-4983-98b9-9203637bcdf1.png)

Youtube menu bottomsheet:
![image](https://user-images.githubusercontent.com/20476002/129249745-eb6e8a52-d7fd-4ec8-a305-9915248a1cc3.png)

Youtube preview:
![image](https://user-images.githubusercontent.com/20476002/129263929-4c0162c8-5cd6-4ae9-b65f-4389fd4e654b.png)

NewPipe:
![image](https://user-images.githubusercontent.com/20476002/129247066-9c3e233b-48fc-454f-bceb-418307d7dbd7.png)

NewPipe preview:
![image](https://user-images.githubusercontent.com/20476002/129263986-81da2e62-38d8-462f-878d-9049ee02b63a.png)

**Comment by Angelk90** [comment]
@Stypox : How about changing the video preview ui.

At the moment:
![image](https://user-images.githubusercontent.com/20476002/129478015-48652b93-3b4e-48ed-b9e5-7322cd22a822.png)

I put a white border in the preview image and changed the order to the first image and then under the time.
How does youtube.

Change:
![image](https://user-images.githubusercontent.com/20476002/129478033-d278ac49-dc1e-4516-9744-f65e5be27c9b.png)

**Comment by Stypox** [comment]
The time was put above after a decision by the team, and won't be reverted

**Comment by Angelk90** [comment]
@Stypox :About that https://github.com/TeamNewPipe/NewPipe/issues/1317#issuecomment-898258846

**Comment by Thewisem** [comment]
Yeah I want to add an UI design on the chapter element
You could use a line break to denote that that chapter ks finished and this is the next chapter
Or
In the preview you get when you scroll through the playbar, the chapter title appears on top of the preview.
This helps in normal view and we don't need to worry about adding a button in popup view.

**Comment by Pentaphon** [comment]
The closer we can get to VLC player functionality but without the clutter, the better.

**Comment by WhiteChairFromIkea** [comment]
Hey, do you have fullscreen button in exi player? Both in horizontal and landscape modes?
Do I have a bug here (0.27.1)?
![Screenshot_2024-07-14-10-38-29-597_org schabi newpipe](https://github.com/user-attachments/assets/395fcb50-4667-491d-bdbb-8b26a534cefc)
![Screenshot_2024-07-14-10-38-22-831_org schabi newpipe](https://github.com/user-attachments/assets/2f5988ec-72de-4521-9729-6c7d4906210b)
![Screenshot_2024-07-14-10-38-16-244_org schabi newpipe](https://github.com/user-attachments/assets/45066df5-9e80-4d21-9e39-ed57d46d8bc9)

**Comment by sosasees** [comment]
the fullscreen button appears only if it makes sense.

it always appears on a bigger screen like a tablet
but it does not always appear on a small screen like a phone.

on a small screen:
- the fullscreen button appears always if the video is in portrait aspect ratio
- if the video is in landscape aspect ratio, the fullscreen button appears only if "rotate screen" is turned off
  because rotating the screen is the only way to switch between fullscreen mode and partialscreen mode

**Comment by WhiteChairFromIkea** [comment]
> the fullscreen button appears only if it makes sense.
> 
> it always appears on a bigger screen like a tablet but it does not always appear on a small screen like a phone.
> 
> on a small screen:
> 
> * the fullscreen button appears always if the video is in portrait aspect ratio
> * if the video is in landscape aspect ratio, the fullscreen button appears only if "rotate screen" is turned off
>   because rotating the screen is the only way to switch between fullscreen mode and partialscreen mode

Omg, I was so annoyed not having the fullscreen button  anytime thought creating a feature request. It seems I can force it to be always shown by turning on "tabblet mode" in settings. My issue is solved now, thank you! 🙂

**Comment by theScrabi** [comment]
Please be aware that further discussions should better be made over at the NewPlayer repo: https://github.com/teamNewPipe/newplayer

**Comment by sosasees** [comment]
no, NewPlayer will not replace NewPipe.

NewPlayer will be the updated media&nbsp;player component in NewPipe but NewPlayer can also be used in any app
